### ISSUE TITLE
Saving Recipe never finishes (caught by `Save Recipe Test #acceptance` failure)

### ISSUE THREAD

**Issue by martinpitt** [weldr/welder-web#169]
This particularly test fails very often:
```
❱❱❱ jest --verbose -i -t 'Save Recipe Test #acceptance'
 FAIL  test/test_editRecipe.js (15.147s)
  ● Edit Recipe Page › Single Word Recipe Name Scenario › Save Recipe Test #acceptance › should have toast notification pop up when Save button clicked @edit-recipe-page

    Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      
      at pTimeout (node_modules/jest-jasmine2/build/queueRunner.js:53:21)
      at Timeout.callback [as _onTimeout] (node_modules/jsdom/lib/jsdom/browser/Window.js:523:19)
      at ontimeout (timers.js:471:11)
      at tryOnTimeout (timers.js:306:5)
      at Timer.listOnTimeout (timers.js:266:5)

  Edit Recipe Page
    Single Word Recipe Name Scenario
      Menu Nav Bar Check #acceptance
        ○ skipped 1 test
      Title Bar Check #acceptance
        ○ skipped 2 tests
      Create Composition Test For ostree #acceptance
        ○ skipped 2 tests
      Create Composition Test For iso #acceptance
        ○ skipped 2 tests
      Create Composition Test For ami #acceptance
        ○ skipped 2 tests
      Create Composition Test For qcow2 #acceptance
        ○ skipped 2 tests
      Create Composition Test For vhdx #acceptance
        ○ skipped 2 tests
      Export Recipe To Manifest Test #acceptance
        ○ skipped 4 tests
      Save Recipe Test #acceptance
        ✕ should have toast notification pop up when Save button clicked @edit-recipe-page (15018ms)
```

This reproduces locally and also affects Travis CI runs ([example](https://travis-ci.org/weldr/welder-web/jobs/296207007)), so that otherwise valid PRs have to be force-pushed very often to re-trigger the tests.

**Comment by martinpitt** [comment]
When adding the `show: true` option to nightmare, the failure can be watched interactively; the developer console shows this oops:
```
main.5d0c66a767b14f997f0e.js:68 Uncaught TypeError: Cannot read property 'name' of undefined
    at e.value (http://localhost:9090/cockpit/@localhost/welder/dist/main.5d0c66a767b14f997f0e.js:68:15372)
    at t.value (http://localhost:9090/cockpit/@localhost/welder/dist/main.5d0c66a767b14f997f0e.js:72:28042)
    at Object.r (http://localhost:9090/cockpit/@localhost/welder/dist/main.5d0c66a767b14f997f0e.js:34:4114)
    at a (http://localhost:9090/cockpit/@localhost/welder/dist/main.5d0c66a767b14f997f0e.js:34:2451)
    at Object.s [as executeDispatchesInOrder] (http://localhost:9090/cockpit/@localhost/welder/dist/main.5d0c66a767b14f997f0e.js:34:2666)
    at p (http://localhost:9090/cockpit/@localhost/welder/dist/main.5d0c66a767b14f997f0e.js:33:31098)
    at m (http://localhost:9090/cockpit/@localhost/welder/dist/main.5d0c66a767b14f997f0e.js:33:31224)
    at Array.forEach (native)
    at n (http://localhost:9090/cockpit/@localhost/welder/dist/main.5d0c66a767b14f997f0e.js:34:4580)
    at Object.processEventQueue (http://localhost:9090/cockpit/@localhost/welder/dist/main.5d0c66a767b14f997f0e.js:34:260)
[...]
```
This isn't particularly helpful, but it shows that it's not just a test race condition, but the test triggers some actual bug.

**Comment by martinpitt** [comment]
I can actually reproduce this error easily: Either on your local instance or on e. g. http://devel.weld.lab.eng.bos.redhat.com/#/edit/lol, just click on "Edit Recipe" and immediately click on "Save" - the "lol - Saving Recipe" notification bubble pops up, but it never finishes. This is exactly what the integration test does, i. e. it does not actually do any changes to the recipe.

**Comment by martinpitt** [comment]
@jkozol: Is that fixed by your Saving rework in PR #162 by any chance?

**Comment by jgiardino** [comment]
@martinpitt is this a duplicate of #145?

**Comment by martinpitt** [comment]
@jgiardino : No, I don't think so. The stack trace is different, and this happens with a non-empty recipe and with an API server running.

**Comment by jkozol** [comment]
I believe this should be fixed in PR #162 since I am unable to replicate it in the PR. I think this error might be due to us rehydrating the state on page load as well as fetching all of the recipes and the contents of the recipe being edited. This can cause the save recipe API call to be pointed at a recipe state that has been changed so this.recipe may be undefined. 

> I can actually reproduce this error easily: Either on your local instance or on e. g. http://devel.weld.lab.eng.bos.redhat.com/#/edit/lol, just click on "Edit Recipe" and immediately click on "Save" - the "lol - Saving Recipe" notification bubble pops up, but it never finishes. This is exactly what the integration test does, i. e. it does not actually do any changes to the recipe.

I believe these steps reproduce #145 which is fixed in PR #162

**Comment by martinpitt** [comment]
This is now fixed with @jkozol's saving rework, the Save button is disabled while the recipe is empty.
